go/vt/tabletserver/codex.go

```diff
@@ -96,19 +97,43 @@ func resolveListArg(col *schema.TableColumn, key string, bindVars map[string]int
 	if err != nil {
 		return nil, NewTabletError(vtrpcpb.ErrorCode_BAD_INPUT, "%v", err)
 	}
-	list := val.([]interface{})
-	resolved := make([]sqltypes.Value, len(list))
-	for i, v := range list {
-		sqlval, err := sqltypes.BuildConverted(col.Type, v)
-		if err != nil {
-			return nil, NewTabletError(vtrpcpb.ErrorCode_BAD_INPUT, "%v", err)
+
+	switch list := val.(type) {
+	case []interface{}:
+		resolved := make([]sqltypes.Value, len(list))
+		for i, v := range list {
+			sqlval, err := sqltypes.BuildConverted(col.Type, v)
+			if err != nil {
+				return nil, NewTabletError(vtrpcpb.ErrorCode_BAD_INPUT, "%v", err)
+			}
+			if err = validateValue(col, sqlval); err != nil {
+				return nil, err
+			}
+			resolved[i] = sqlval
 		}
-		if err = validateValue(col, sqlval); err != nil {
-			return nil, err
+		return resolved, nil
+	case *querypb.BindVariable:
+		if list.Type != querypb.Type_TUPLE {
+			return nil, NewTabletError(vtrpcpb.ErrorCode_BAD_INPUT, "expecting list for bind var %s: %v", key, list)
+		}
+		resolved := make([]sqltypes.Value, len(list.Values))
+		for i, v := range list.Values {
+			// FIXME(alainjobart,sougou): we ignore the
+			// input's v.Type in this expression. I am not
+			// sure this is the way to go.
+			sqlval, err := sqltypes.BuildConverted(col.Type, v.Value)
```

Comment:
All you have to do is pass this in instead: `sqltypes.MaleTrusted(v.Typ, v.Value)`.
Then BuildConverted will `do the right thing`.